### DISCUSSION TITLE
How to allow a Blueprint's taxonomy term field to autocreate a non-defined term

### DISCUSSION THREAD

**Discussion by ryanvelbon** [statamic/cms#6389] [Q&A] [answered]
**TLDR**
Can Statamic automatically generate a `content/taxonomies/tags/foo.yaml` file when a new tag "foo" is added to my article's tags array?

-------------------

Hey all.

This is my first attempt at building a generic blog website using statamic. I have a `Post `blueprint (as in blog post).

As is typical in CMS's like WordPress, I would like that a post can have tags and categories.

Take the example post entry below:

```
---
id: e948a65f-8b90-4371-841c-ae1863d4cb4e
published: false
blueprint: posts
title: 'The Best Fruit in the World'
tags:
  - apple
  - banana
  - cherry
---
```

Suppose I haven't yet created the Tag entry `cherry` and I nonetheless go ahead and tag `cherry` to the tags array in my post's Front Matter, how do I make Statamic auto-create a `content/taxonomies/tags/cherry.yaml` file.

The `Tag` blueprint essentially just consists of a title field.

**Comment by ryanmitchell** [comment]
Im fairly sure this is what happens? If you add a new term in an entry editing view it will be created for you when you have "Allow creating" enabled in the taxonomy field config.

**Comment by jasonvarga** [comment, marked as answer]
It will only create the term file if you add the term to the entry through the control panel.

If you just add it directly to the front-matter, a file won't be created. However, it will still function as a term.

**Reply by ryanvelbon** [reply]
That's kind of a bummer. Hopefully in a future release there will be a command like `php please generate-entries --taxonomy=tags` or something which would go through all collection entries, scan the Front Matter and generate a `/content/taxonomies/tags/$tag.yaml` file for any new tag.

**Reply by jasonvarga** [reply]
Everything still functions, there's literally just no file.

If you go into the taxonomy section in the cp, you can edit the term. When you save, a file will be created.
